### PR TITLE
Add support for optional node data

### LINKED ISSUE #12: Extend the library to be able to add data to nodes
Hi there,

I have a feature request. I was wondering if the library could be extended to enable the user to add a node with data

From what I can tell this would require 3 changes:

* the addNode method needs a 2nd parameter for the data (first parameter would be the node name and second parameter the data)
* a new method getData will be needed to get the data of a node by node name
* the hasNode method would need changes to be able to cope with falsy data (0, '', false, undefined, null)

Backwards compatibility should not be an issue if we keep the existing behavior when the addNode method gets undefined data. Only issue with this would be that a user will not be able to set undefined as data for a node.

In case you are interested in this functionality, I would gladly prepare a pull request.

Cheers
Nikolas

### PR DESCRIPTION
**Pull request by nponiros**
Add optional data parameter for the addNode method. If data is not given
the node name is used as data.

Add methods setNodeData and getNodeData to manipulate the data
associated with a node.

Update the hasNode method to be able to cope with falsy node data.

Add tests for new functionality.

Update README to document the new functionality.

Fixes #12

### CODE REVIEW

File: lib/dep_graph.js [hunk 1]
```diff
@@ -74,7 +79,25 @@ DepGraph.prototype = {
...
+    if (this.hasNode(node)) {
+      return this.nodes[node];
+    } else {
+      throw new Error('Node does not exist: ' + node);
+    }
+  },
+  /**
+   * Set the associated data for a given node name. If the node does not exist, this method will do nothing
+   */
+  setNodeData:function (node, data) {
```
Comment (jriecken) [inline lib/dep_graph.js]: I feel like this should also throw an error if the node doesn't exist. `setNodeData` shouldn't be called on non-existent nodes.

**Comment by jriecken** [comment]
Just one minor comment - I'll merge it after it's fixed 😄

**Comment by jriecken** [comment]
Published new version to npm - 0.5.0

**Comment by nponiros** [comment]
Thanks for the merge and the new release.

**Pull request merged**